events/api.py

```diff
@@ -2578,76 +2578,64 @@ def _filter_event_queryset(queryset, params, srs=None):  # noqa: C901
             local=True,
         )
 
-    val = params.get("local_ongoing_OR", None)
-    if val:
-        rc = _terms_to_regex(val, "OR")
-        ids = {
-            k
-            for k, v in cache.get("local_ids").items()
-            if rc.search(v, concurrent=True)
-        }
-        queryset = queryset.filter(id__in=ids)
+    def get_queryset_from_cache(params, param, cache_name, operator, queryset):
+        cache_values = cache.get(cache_name)
+        if cache_values:
+            val = params.get(param, None)
+            if val:
+                rc = _terms_to_regex(val, operator)
+                ids = {
+                    k for k, v in cache_values.items() if rc.search(v, concurrent=True)
+                }
+                queryset = queryset.filter(id__in=ids)
```

Comment:
Could do this to reduce nesting:
```suggestion
    def get_queryset_from_cache(params, param, cache_name, operator, queryset):
        cache_values = cache.get(cache_name)
        val = params.get(param, None)
        if not val or not cache_values:
            return queryset

        rc = _terms_to_regex(val, operator)
        ids = {
             k for k, v in cache_values.items() if rc.search(v, concurrent=True)
        }
        return queryset.filter(id__in=ids)
```
Makes it a bit easier to read. Same for the other function, `get_queryset_from_cache_many`.

Reply:
Changed as suggested.